internal/worker/remoterelationconsumer/remoterelations/worker.go

```diff
@@ -145,58 +140,99 @@ func (w *remoteRelationsWorker) loop() error {
 
 	// Totally new so start the lifecycle watcher.
 	watcher, err := w.client.WatchRelationSuspendedStatus(ctx, params.RemoteEntityArg{
-		Token:         w.localRelationToken,
+		Token:         w.consumerRelationUUID.String(),
 		Macaroons:     macaroon.Slice{w.macaroon},
 		BakeryVersion: bakery.LatestVersion,
 	})
 	if err != nil {
-		return errors.Annotatef(err, "watching remote side of relation for %q", w.relationTag)
+		return errors.Annotatef(err, "watching offerer side of relation for %q", w.consumerRelationUUID)
 	}
 
 	if err := w.catacomb.Add(watcher); err != nil {
-		return errors.Annotatef(err, "adding remote relation status watcher for %q", w.relationTag)
+		return errors.Annotatef(err, "adding offerer relation status watcher for %q", w.consumerRelationUUID)
 	}
 
+	var event RelationChange
 	for {
 		select {
 		case <-w.catacomb.Dying():
 			return w.catacomb.ErrDying()
 
 		case changes, ok := <-watcher.Changes():
 			if !ok {
-				// We are dying.
-				return w.catacomb.ErrDying()
+				select {
+				case <-w.catacomb.Dying():
+					return w.catacomb.ErrDying()
+				default:
+					return errors.NotValidf("relation status watcher closed unexpectedly for %q", w.consumerRelationUUID)
+				}
 			}
 			if len(changes) == 0 {
 				w.logger.Warningf(ctx, "relation status watcher event with no changes")
 				continue
 			}
 
 			// We only care about the most recent change.
+			// We must ensure that all changes being sent must be sent in order.
 			change := changes[len(changes)-1]
-			w.logger.Debugf(ctx, "relation status changed for %v: %v", w.relationTag, change)
-
-			event := RelationChange{
-				Tag:                     w.relationTag,
-				RelationToken:           w.remoteRelationToken,
-				ApplicationOrOfferToken: w.applicationToken,
-				Life:                    change.Life,
-				Suspended:               change.Suspended,
-				SuspendedReason:         change.SuspendedReason,
+			w.logger.Debugf(ctx, "relation status changed for %v: %v", w.consumerRelationUUID, change)
+
+			event = RelationChange{
+				ConsumerRelationUUID:   w.consumerRelationUUID,
+				OffererApplicationUUID: w.offererApplicationUUID,
+				Life:                   change.Life,
+				Suspended:              change.Suspended,
+				SuspendedReason:        change.SuspendedReason,
 			}
 
 			select {
 			case <-w.catacomb.Dying():
 				return w.catacomb.ErrDying()
 			case w.changes <- event:
 			}
+
+		case req := <-w.requests:
```

Comment:
I think this block warrants a comment on the `requests` member, and what this is for. It's essentially facilitating the report without locks, but let's be clear to the reader.